src/Dialect/Mlir/DialectBuilder.cpp

```diff
@@ -695,7 +697,7 @@ TypedAttr MathBuilder::negativeInfAttr(Type type) const {
         default:
           llvm_unreachable("unsupported element type");
         }
-        attr = b().getIntegerAttr(type, APInt(width, value));
+        attr = b().getIntegerAttr(type, APInt(width, value, false, true));
```

Comment:
I had to update the `APInt` with the following boolean values based on the this PR update: https://github.com/llvm/llvm-project/pull/114539

Previously truncation was set to true as default and they changed it to false, we need the truncation on for our lit test to pass and to be consistent with the previous behavior.